### PR TITLE
Fix #7460

### LINKED ISSUE #7460: Unrecognised target ':myfirstapp' when following getting started guide
**Describe the bug**

While following the [getting started guide](https://cabal.readthedocs.io/en/3.4/getting-started.html#getting-started-with-haskell-and-cabal) I am running into an error message in the section [running the application](https://cabal.readthedocs.io/en/3.4/getting-started.html#running-the-application).

**To Reproduce**

```
$ mkdir myfirstapp
$ cd myfirstapp
$ cabal init --cabal-version=2.4 --license=NONE -p myfirstapp
...
$ cabal v2-run :myfirstapp
cabal: Unrecognised target ':myfirstapp'.
Expected a meta-target 'all', rather than 'myfirstapp'.
```

**Expected behavior**

Building and running the 'myfirstapp' application without errors.

**System information**
 - Operating system: NixOS 21.05
 - `cabal` 3.4.0.0, `ghc` 8.10.4

### PR DESCRIPTION
**Pull request by noughtmare**
---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [x] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: doc/getting-started.rst [hunk 1]
```diff
@@ -70,25 +70,22 @@ As mentioned above, ``cabal init`` with no arguments generates a package with a
...
 .. code-block:: console
 
-     $ cabal run :myfirstapp
+     $ cabal run myfirstapp
      ...
      Hello, Haskell!
 
-.. note:: The ``:`` prefix in ``:myfirstapp`` signifies that the
-	  ``myfirstapp`` target is part of the current package.
-
```
Comment (fgaz) [inline doc/getting-started.rst:85]: I wonder what was meant by this. @m-renaud did it work in the past or did you intend to write `.:myfirstapp`?

**Review by fgaz — APPROVED** [review]
Looks good!

**Review by JonathanLorimer — APPROVED** [review]

File: doc/getting-started.rst (hunk 1 above)
Comment (emilypi) [reply doc/getting-started.rst:85]: This is actually accepted syntax 💀 

```shell
λ P haskell → mkdir tlkajdfd
λ P haskell → cd tlkajdfd 
λ P tlkajdfd → cabal init

Guessing dependencies...

Generating LICENSE...
Warning: unknown license type, you must put a copy in LICENSE yourself.
Generating CHANGELOG.md...
Generating app/Main.hs...
Generating tlkajdfd.cabal...

Warning: no synopsis given. You should edit the .cabal file and add one.
You may want to edit the .cabal file and add a Description field.
λ P tlkajdfd → ls
CHANGELOG.md   app            tlkajdfd.cabal
λ P tlkajdfd → cabal run :tlkajdfd
Resolving dependencies...
Build profile: -w ghc-8.10.5 -O1
In order, the following will be built (use -v for more details):
 - tlkajdfd-0.1.0.0 (exe:tlkajdfd) (first run)
Configuring executable 'tlkajdfd' for tlkajdfd-0.1.0.0..
Preprocessing executable 'tlkajdfd' for tlkajdfd-0.1.0.0..
Building executable 'tlkajdfd' for tlkajdfd-0.1.0.0..
[1 of 1] Compiling Main             ( app/Main.hs, /Users/emilypi/haskell/tlkajdfd/dist-newstyle/build/x86_64-osx/ghc-8.10.5/tlkajdfd-0.1.0.0/x/tlkajdfd/build/tlkajdfd/tlkajdfd-tmp/Main.o )
Linking /Users/emilypi/haskell/tlkajdfd/dist-newstyle/build/x86_64-osx/ghc-8.10.5/tlkajdfd-0.1.0.0/x/tlkajdfd/build/tlkajdfd/tlkajdfd ...
Hello, Haskell!
λ P tlkajdfd → 
```

Yeah no - let's not introduce people to that.

**Review by emilypi — APPROVED** [review]
LGTM - thanks!

File: doc/getting-started.rst (hunk 1 above)
Comment (noughtmare) [reply doc/getting-started.rst:85]: @emilypi I'm getting the same error as in #7460 when running your commands with cabal-install and Cabal-the-library both version 3.4.0.0.

Comment (emilypi) [reply doc/getting-started.rst:85]: Weird. I'm running the same: 

```
λ P tlkajdfd → cabal --version
cabal-install version 3.4.0.0
compiled using version 3.4.0.0 of the Cabal library 
λ P tlkajdfd → ghc --version
The Glorious Glasgow Haskell Compilation System, version 8.10.5
λ P tlkajdfd → 
```

What distro?

Comment (noughtmare) [reply doc/getting-started.rst:85]: I'm running NixOS 21.05 and GHC 8.10.4.

Comment (emilypi) [reply doc/getting-started.rst:85]: Ahh. I'm on Macos, and it works for me. I can get to my ubuntu box in a few days, but I wonder if @m-renaud wrote this while on a Mac!

**Comment by fgaz** [comment]
@emilypi doesn't work for me! (using recent-ish master)

```
$  mkdir tlkajdfd
$  cd tlkajdfd/
$  cabal init --non-interactive 

Guessing dependencies...

Using cabal specification: 3.0

Warning: unknown license type, you must put a copy in LICENSE yourself.
Creating CHANGELOG.md...
Creating file CHANGELOG.md...
Creating directory ./app...
Creating file app/Main.hs...
Creating file tlkajdfd.cabal...

Warning: no synopsis given. You should edit the .cabal file and add one.
You may want to edit the .cabal file and add a Description field.
$  ls
app  CHANGELOG.md  tlkajdfd.cabal
$  cabal run :tlkajdfd
cabal: Unrecognised target ':tlkajdfd'.
Expected a meta-target 'all', rather than 'tlkajdfd'.
```

**Comment by emilypi** [comment]
@fgaz you on Linux too?

**Comment by fgaz** [comment]
Yes, NixOS, but cabal is built with cabal

**Comment by emilypi** [comment]
A mystery indeed.

**Comment by fgaz** [comment]
Opened #7474 to track it

**Comment by jneira** [comment]
Maybe a regression test over this will be needed?

**Comment by Mikolaj** [comment]
@jneira: regression test for #7474 or for the documentation fix or for something else altogether?

**Comment by jneira** [comment]
for `cabal run :myexecomp` and all os's, not sure if it used to work or doesn't but a regression test would alert us

**Comment by Mikolaj** [comment]
Oh, yes, sure, I was confused by the idea of a regression test for a feature that is broken --- the test would always fail. :) Sure, once it's fixed, we need a regression test.

**Comment by fgaz** [comment]
I think it can also be added _before_ fixing it. iirc there's a way to mark a test as broken referencing a ticket

**Comment by Mikolaj** [comment]
Aha! TDD! I like it.

**Pull request merged**